src/picongpu/include/plugins/adios/restart/LoadSpecies.hpp

```diff
@@ -126,19 +125,31 @@ struct LoadSpecies
         ADIOS_CMD(adios_perform_reads( params->fp, 1 ));
         adios_selection_delete(piSel);
 
-        /* search my entry (using my scalar position) in particlesInfo */
+        /* Run a prefix sum over the numParticles[0] element in particlesInfo
+         * to retreive the offset of particles before gc.getScalarPosition() */
         uint64_t particleOffset = 0;
         uint64_t myScalarPos = gc.getScalarPosition();
 
+        /** \todo in a world where gc.getScalarPosition() is actually an MPI rank
+         * the log(N) scaling Exscan could be used ...
+         * MPI_CHECK(MPI_Exscan( &totalNumParticles, &particleOffset, 1, MPI_UINT64_T, MPI_SUM, params->adiosComm ));
+         * Rank 0 needs special treatmend in `MPI_Exscan`
+         * if( gc.getScalarPosition() == 0 )
+         *    particleOffset = 0; */
+
+        uint64_t oneParticlesInfo[2] = { particlesInfo[0], myScalarPos };
+        uint64_t fullParticlesInfo[2 * gc.getGlobalSize()];
+
+        MPI_CHECK(MPI_Allgather( oneParticlesInfo, 2, MPI_UINT64_T,
+                                 fullParticlesInfo, 2 * gc.getGlobalSize(), MPI_UINT64_T,
+                                 params->adiosComm ));
+
         for (size_t i = 0; i < gc.getGlobalSize(); ++i)
         {
-            if (particlesInfo[i][1] == myScalarPos)
-            {
-                totalNumParticles = particlesInfo[i][0];
-                break;
-            }
-
-            particleOffset += particlesInfo[i][0];
+            if( fullParticlesInfo[i*2 + 1] < myScalarPos )
```

Comment:
the main problem here is, that the offSet in [ADIOSCountParticles.hpp:120](https://github.com/ComputationalRadiationPhysics/picongpu/blob/f0ba5151411ac09eac317f97b5d93e05e20ce29f/src/picongpu/include/plugins/adios/ADIOSCountParticles.hpp#L120-L125) that is used to write particles in the global offset is calculated *not* by ScalarPos but by rank (which can potentially change and break).

We therefore now simply read and allgather the local domain offset and compare with that.